### ISSUE TITLE
Allow Hermes to relay `wasm` messages

### ISSUE THREAD

**Issue by ljoss17** [informalsystems/hermes#2809]
## Summary

Hermes only detects `ibc*` events. By detecting `wasm` events it would be possible to relay `wasm` messages.

## Acceptance Criteria

- [ ] Hermes can relay `wasm` messages

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate milestone (priority) applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by ancazamfir** [comment]
I think the solution here is to allow the configuration of desired `message.module` query strings.

**Comment by alpe** [comment]
I have found this issue while researching the neutron wasmd [patch](https://github.com/neutron-org/wasmd/commit/297bdc7c582365ef60a45def4e7c631cbe5d2836#diff-171d7d96b1521ea0c90ca4c1d112065104dab4d6c1219f84f60a5c2dc3d28d6dR204-R205). Now I have some better context for the patch but I am wondering what the best solution would be?

In wasmd we keep only the root "message" event that started the process. Therefore "close channel" or ics-20 "transfer" messages emitted by a contract will not show up. 

In my current understanding, hermes is looking at the `message` events emitted from ibc module operations for relayer work.
For example `message.module = ibc_connection` would cover 
`connection_open_init || connection_open_try || connection_open_ack || connection_open_confirm` operations. 
So I am thinking that it may make more sense to not rely on "message" events and "module" attribute but use the concrete events defined in [ibc-go](https://github.com/cosmos/ibc-go/blob/main/docs/ibc/events.md). 
By doing this, the event queries would become a bit more complex but there would not be any need to make this "customizable" or patch wasmd.

**Comment by romac** [comment]
> So I am thinking that it may make more sense to not rely on "message" events and "module" attribute but use the concrete events defined in [ibc-go](https://github.com/cosmos/ibc-go/blob/main/docs/ibc/events.md).

@alpe I am not sure I understand what you mean by that? Could expand a bit on that?

**Comment by romac** [comment]
> In wasmd we keep only the root "message" event that started the process. Therefore "close channel" or ics-20 "transfer" messages emitted by a contract will not show up.
> So I am thinking that it may make more sense to not rely on "message" events and "module" attribute but use the concrete events defined in [ibc-go](https://github.com/cosmos/ibc-go/blob/main/docs/ibc/events.md).

@alpe I am not sure I understand what you mean here, could you please expand a bit on what's the problem and what you would suggest we do instead?

**Comment by alpe** [comment]
tldr: there is no ibc ['message' event](https://github.com/informalsystems/hermes/blob/v1.3.0/crates/relayer/src/event/monitor.rs#L158) when the channel was created via CosmWasm [submessage](https://github.com/CosmWasm/wasmd/blob/v0.31.0-rc2/x/wasm/keeper/msg_dispatcher.go#L195). Same for IBC packet send.

If hermes would query the non "message" [events types](https://github.com/cosmos/ibc-go/blob/v7.0.0-rc1/docs/ibc/events.md) instead then it should all work nicely.

**Comment by ancazamfir** [comment]
@alpe is it possible to show us and example of the `/block_results` output for a block that includes the wasm emitted events, for example for packets sent/received?

**Comment by alpe** [comment]
@ancazamfir I don't have an example. 
But as I tried to outline before: don't look at the wasm events but the [ibc-go](https://github.com/cosmos/ibc-go/blob/main/docs/ibc/events.md) instead.
